core/lib/makeswift/components/carousel/carousel.makeswift.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Checkbox,
+  List,
+  Link as MSLink,
+  Select,
+  Shape,
+  Slot,
+  Style,
+  TextInput,
+} from '@makeswift/runtime/controls';
+
+import {
+  Carousel,
+  CarouselButtons,
+  CarouselContent,
+  CarouselItem,
+  CarouselScrollbar,
+} from '@/vibes/soul/primitives/carousel';
+import { Link } from '~/components/link';
+import { runtime } from '~/lib/makeswift/runtime';
+
+interface Slide {
+  link: { href?: string; target?: string };
+  children: React.ReactNode;
+}
+
+interface MSCarouselProps {
+  className: string;
+  slides?: Slide[];
+  showScrollbar: boolean;
+  showArrows: boolean;
+  colorScheme: 'light' | 'dark';
+}
+
+runtime.registerComponent(
+  function MSCarousel({
+    className,
+    slides,
+    showScrollbar = true,
+    showArrows = true,
+    colorScheme,
+  }: MSCarouselProps) {
+    return (
+      <div className={className}>
+        {!slides || slides.length < 1 ? (
+          <div className="p-4 text-center text-lg text-gray-400">Add items to the carousel</div>
+        ) : (
+          <Carousel>
+            <CarouselContent className="mb-20">
+              {slides.map(({ link, children }, index) => (
+                <CarouselItem
+                  className="basis-full @md:basis-1/2 @lg:basis-1/3 @2xl:basis-1/4"
+                  key={index}
+                >
+                  <Link href={link.href ?? ''} target={link.target ?? ''}>
```

Comment:
Should we default this to `#` when there's no `href`? As for `target`, we should set it either to `_self`, the default or `undefined` so it's not set to an empty string, which isn't a valid value